pkg/build/builder/dockerutil.go

```diff
@@ -39,15 +47,31 @@ func pushImage(client DockerClient, name string, authConfig docker.AuthConfigura
 		opts.OutputStream = os.Stderr
 	}
 	var err error
-	for retries := 0; retries <= DefaultPushRetryCount; retries++ {
-		err = client.PushImage(opts, authConfig)
+
+	for retries := 0; retries <= MaxPushRetryCount; retries++ {
+
+		err := client.PushImage(opts, authConfig)
 		if err == nil {
 			return nil
 		}
-		if retries == DefaultPushRetryCount {
+
+		errMessage := fmt.Sprintf("%s", err)
+		var errInfo string
+
+		switch true {
+		case strings.Contains(errMessage, "v1 ping attempt failed with error"):
+			if retries > DefaultPushRetryCount {
+				return err
+			}
+			errInfo = " cause Docker registry is unavailable,"
+		case strings.Contains(errMessage, "is already in progress"):
+			errInfo = " cause other image is already being pushed to registry,"
+		default:
 			return err
 		}
-		util.HandleError(fmt.Errorf("push for image %s failed, will retry in %s ...", name, DefaultPushRetryDelay))
+
+		util.HandleError(fmt.Errorf("push for image %s failed,%s will retry in %s ...", name, errInfo, DefaultPushRetryDelay))
```

Comment:
didnt wanted to add it there cause if the push fails due to other case then we are trying to catch here, there would be two whitespaces.
Instead I've add the extra whitespace in the `errInfo` variable